plugins/filters/app_janus/httpSender.js

```diff
@@ -0,0 +1,184 @@
+                    sending.map(s => ({...s, retries: s.retries + 1})))
+                fireMetrics()
+                waitingForResend -= sending.length
+                retransmissions += sending.length
+            }, 10000)
+        } finally {
+            sendingQueueLength -= sendingLength
+            httpThroughputMetrics[Math.floor(Date.now()/1000) % 5] += body.length
+        }
+        await new Promise(f => setTimeout(f, 100))
+    }
+}
+
+async function startSpansSender () {
+    while (true) {
+        if (!spansToSend.length) {
+            await new Promise(f => onSpans.push(f))
+        }
+        let toSend = ['[']
+        let len = 2
+        let i = 0
+        spansToSend = spansToSend.filter(m => m.payload)
+        for (; i < spansToSend.length && len < 20 * 1024 * 1024; i++) {
+            if (i > 0) {
+                toSend.push(',')
+            }
+            let str = JSON.stringify(spansToSend[i].payload)
+            if (!str) {
+                continue
+            }
+            str = str.substring(1, str.length - 1)
+            toSend.push(str)
+            len += str.length
+        }
+        toSend.push(']')
+        let sending = spansToSend.slice(0, i)
+        spansToSend = spansToSend.slice(i)
+        fireSpans()
+        const body = toSend.join('')
+        toSend = null
+        const sendingLength = sending.length
+        sendingQueueLength += sendingLength
+        sending = sending.filter(s => s.retries < 4)
+        const givingUp = sendingLength - sending.length
+        try {
+            var response = await axios.post(`${module.exports.host}/tempo/spans`, body, {
+                maxBodyLength: 50 * 1024 * 1024,
+                maxContentLength: 50 * 1024 * 1024,
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            })
+            //logger.debug('Spans posted', response.status, response.statusText)
+        } catch (e) {
+            sendingErrors += sendingLength
+            if (e instanceof axios.AxiosError) {
+                logger.error(`HTTP ERROR '${e.message}' [${e.response?.status}]: ${e.response?.data}`)
+            } else {
+                logger.error(e)
+            }
+            waitingForResend += sending.length
+            gaveUp += givingUp
+            setTimeout(() => {
+                spansToSend.push.apply(spansToSend,
+                    sending.filter(s => s.retries < 4).map(s => ({...s, retries: s.retries + 1})))
+                fireSpans()
+                waitingForResend -= sending.length
+                retransmissions += sending.length
+            }, 10000)
+        } finally {
+            sendingQueueLength -= sendingLength
+            httpThroughputMetrics[Math.floor(Date.now()/1000) % 5] += body.length
+        }
+        await new Promise(f => setTimeout(f, 100))
+    }
+}
+
+for (let i = 0; i < 10; i++) {
+    startMetricsSender()
+    startSpansSender()
+}
+
+(() => {
```

Comment:
I would hide this behind a debug flag inside the code. 

try a global debug variable set to false by default, as in production we do not want to show this report constantly.

thoughts?